### PR TITLE
Add fight-club strategy [fight-club]

### PR DESCRIPTION
**Pull request by MantisClone**
Changes proposed in this pull request:
- Add a `fight-club` strategy that calculates the voter's score based on glove NFTs and weight class Kudos.

I've chosen to omit a `schema.json` with this PR because the `"type": "object"` parameters don't render in snapshot playground, so I'd rather have the simple parameter entry box.

### CODE REVIEW

**Comment by MantisClone** [comment]
Huh.. I don't understand why the `test-strategy` workflow keeps getting skipped.  I'll just run the tests locally I guess.

**Comment by ChaituVR** [comment]
Hello @MantisClone looks like it is same as erc20-balance-of? you want to add `calculates the voter's score based on glove color and weight class NFTs.`?

File: src/strategies/arrow-vesting/index.ts [hunk 1]
```diff
@@ -7,79 +7,106 @@ export const author = 'BrassLion';
 export const version = '0.1.0';
 
 const vestingFactoryAbi = [
-  "function escrows_length() public view returns (uint256)",
-  "function escrows(uint256 index) public view returns (address)",
+  'function escrows_length() public view returns (uint256)',
+  'function escrows(uint256 index) public view returns (address)'
```
Comment (ChaituVR) [inline src/strategies/arrow-vesting/index.ts:11]: @MantisClone Make sure there are no changes in other strategies when we go here https://github.com/snapshot-labs/snapshot-strategies/pull/782/files#

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/arrow-vesting/index.ts (hunk 1 above)
Comment (MantisClone) [reply src/strategies/arrow-vesting/index.ts]: Done.

File: src/strategies/fight-club/index.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      multi.call(
+        `${address}.gloves.${gloveAddress}`,
+        gloveAddress,
+        'balanceOf(address)',
+        [address]
+      );
+    });
+  });
+
+  Object.entries(options.weightClassIds).forEach(([weightClassId]) => {
```
Comment (ChaituVR) [inline src/strategies/fight-club/index.ts:35]: Hi @MantisClone Sorry for not noticing this before, but I feel we should add a limit to `gloveAddresses` and `weightClassIds` if someone passes unlimited entries, there could be memory issues. wyt?

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/fight-club/index.ts (hunk 2 above)
Comment (MantisClone) [reply src/strategies/fight-club/index.ts:35]: I'd be okay with adding a limit. I've decided to arbitrarily limit it to 10 glove colors and 10 weight classes. 

If I had unlimited time, I'd run a test to determine the actual limits. But in practice, I know there will only be 4 glove colors and 4 or 5 weight classes. So setting an arbitrary limit is the more pragmatic approach at this time.

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**